examples/tv-casting-app/android/App/app/src/main/java/com/chip/casting/app/CommissionerDiscoveryFragment.java

```diff
@@ -8,143 +8,183 @@
+        list.setOnItemClickListener((parent, view1, position, id) -> {
+            DiscoveredNodeData discoveredNodeData = (DiscoveredNodeData)parent.getItemAtPosition(position);
+            Log.d(
+                    TAG,
+                    "CommissionerResolveListener.onServiceResolved.OnClickListener.onClick called for "
+                    + discoveredNodeData);
+            Callback callback1 =
+                (Callback) getActivity();
+            callback1.handleCommissioningButtonClicked(discoveredNodeData);
         });
 
-    Log.d(TAG, "Auto discovering");
-    tvCastingApp.discoverVideoPlayerCommissioners(
-        DISCOVERY_DURATION_SECS, successCallback, failureCallback);
-  }
-
-  @VisibleForTesting
-  public String getCommissionerButtonText(DiscoveredNodeData commissioner) {
-    String main = commissioner.getDeviceName() != null ? commissioner.getDeviceName() : "";
-    String aux =
-        "" + (commissioner.getProductId() > 0 ? "Product ID: " + commissioner.getProductId() : "");
-    aux +=
-        commissioner.getDeviceType() > 0
+        this.successCallback =
+            new SuccessCallback<DiscoveredNodeData>() {
+                @Override
+                public void handle(DiscoveredNodeData discoveredNodeData) {
+                    Log.d(TAG, "Discovered a Video Player Commissioner: " + discoveredNodeData);
+                    new Handler(Looper.getMainLooper())
+                        .post(() -> {
+                            if (commissionablePlayerList.contains(discoveredNodeData)) {
+                                Log.d(TAG, "Replacing existing entry in players list");
+                                arrayAdapter.remove(discoveredNodeData);
+                            }
+                            arrayAdapter.add(discoveredNodeData);
+                        });
+                }
+            };
+
+        this.failureCallback =
+            new FailureCallback() {
+                @Override
+                public void handle(MatterError matterError) {
+                    Log.e(TAG, "Error occurred during video player commissioner discovery: " + matterError);
+                    if (MatterError.ERROR_SERVICE_LOST == matterError.getErrorCode()) {
+                        Log.d(TAG, "Attempting to restart service");
+                        tvCastingApp.discoverVideoPlayerCommissioners(successCallback, this);
+                    }
+                }
+            };
+
+        Button discoverButton = getView().findViewById(R.id.discoverButton);
+        discoverButton.setOnClickListener(
+                v -> {
+                    Log.d(TAG, "Discovering on button click");
+                    tvCastingApp.discoverVideoPlayerCommissioners(successCallback, failureCallback);
+                });
+    }
+
+    @Override
+    public void onResume()
+    {
+        super.onResume();
+        Log.d(TAG, "Auto discovering");
+
+        poller = Executors.newSingleThreadScheduledExecutor()
```

Comment:
It would be better to create the `Executor` once in `onCreate` rather than spinning up a new `Executor` every time the `Fragment` is resumed.